Source/WebCore/loader/cache/CachedResourceLoader.cpp

```diff
@@ -447,10 +498,14 @@ bool CachedResourceLoader::checkInsecureContent(CachedResource::Type type, const
         // These resource can inject script into the current document (Script,
         // XSL) or exfiltrate the content of the current document (CSS).
         if (RefPtr frame = this->frame()) {
-            if (m_document && !MixedContentChecker::frameAndAncestorsCanRunInsecureContent(*frame, m_document->protectedSecurityOrigin(), url))
-                return false;
+            if (m_document) {
+                if (!MixedContentChecker::frameAndAncestorsCanRunInsecureContent(*frame, m_document->securityOrigin(), url))
+                    return false;
+                if (!m_document->settings().upgradeMixedContentEnabled())
+                    return true;
+            }
         }
-        break;
+        FALLTHROUGH;
```

Comment:
FALLTHROUGH makes it hard to read.
Can we just return true?

Reply:
I refactored this part of the code so it is more understandable.